### ISSUE TITLE
unnecessary_override should special case noSuchMethod method

### ISSUE THREAD

**Issue by a14n** [dart-lang/linter#626]
`noSuchMethod` has to be override in some cases:

```dart
abstract class A {
  m();
}
class B implements A {
  @override
  noSuchMethod(Invocation invocation) => super.noSuchMethod(invocation);
}
```

Without the override I get a warning:

>Missing concrete implementation of 'A.m'

**Comment by Hixie** [comment]
I think the solution is to just put @proxy on class B. The @override on noSuchMethod is needed anyway because Object has a noSuchMethod, no?

**Comment by a14n** [comment]
I don't think `@proxy` solve the problem. `@proxy` allows to remove warnings on call site where an unknown method is called.

```dart
abstract class A {
  m();
}


class B implements A { // warning: missing concrete implementation of A.m
}

class WithoutProxy implements A { // no warning: noSuchMethod handle all abstract A methods
  @override
  noSuchMethod(Invocation invocation) => super.noSuchMethod(invocation);
}

@proxy
class WithProxy implements A { // no warning: noSuchMethod handle all abstract A methods
  @override
  noSuchMethod(Invocation invocation) => super.noSuchMethod(invocation);
}

main() {
  new B().m2();
  new WithoutProxy().m2();
  new WithProxy().m2(); // no warning because of @proxy
}
```
